MarvelApiClient/MarvelBaseApiClient.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  BaseApiClient.swift
+//  MarvelApiClient
+//
+//  Created by Pedro Vicente on 14/11/15.
+//  Copyright © 2015 GoKarumi S.L. All rights reserved.
+//
+
+import Foundation
+import BrightFutures
+
+public class MarvelBaseApiClient {
+
+    let baseEndpoint: String
+    let httpClient: HTTPClient
+    let timeProvider: TimeProvider
+
+    init(baseEndpoint: String, timeProvider: TimeProvider, httpClient: HTTPClient) {
+        self.baseEndpoint = baseEndpoint
+        self.timeProvider = timeProvider
+        self.httpClient = httpClient
+    }
+
+    func sendRequest(verb: HTTPVerb, path: String,
+                     params: [String:String]? = [String:String]()) -> Future<HTTPResponse, NSError> {
+        let parameters = addDefaultParams(params)
```

Comment:
And if we user NSURL, we use NSURLComponents tu build the string and and NSURL with it.